src/java/generator/MapGenerator.java

```diff
@@ -20,34 +23,52 @@
 	public static final String VERSION = "0.1.0";
 
 	public static void main(String[] args) throws ExecutionException, InterruptedException {
+
+		String folderPath ="";
+		String version ="";
+		String mapName ="";
+		long seed = 0L;
+
 		try {
-			String folderPath = args[0];
-			long seed = Long.parseLong(args[1]);
-			String version = args[2];
-			String mapName = args.length >= 4 ? args[3] : "NeroxisGen_" + VERSION + "_" + seed;
-			
+			folderPath = args[0];
+			seed = Long.parseLong(args[1]);
+			version = args[2];
+			mapName = args.length >= 4 ? args[3] : "NeroxisGen_" + VERSION + "_" + seed;
+		} catch (ArrayIndexOutOfBoundsException | NumberFormatException e) {
+			System.out.println("Usage: generator [targetFolder] [seed] [expectedVersion] (mapName)");
+			System.exit(1);
+		}
+		finally {
 			if(version.equals(VERSION)) {
 				MapGenerator generator = new MapGenerator();
 				System.out.println("Generating map " + mapName);
 				SCMap map = generator.generate(seed);
 				System.out.println("Saving map to " + Paths.get(folderPath).toAbsolutePath());
 				generator.save(folderPath, mapName, map, seed);
 				System.out.println("Done");
-				
+
 			} else {
 				System.out.println("This generator only supports version " + VERSION);
-				
+
 			}
-		} catch (ArrayIndexOutOfBoundsException | NumberFormatException e) {
-			System.out.println("Usage: generator [targetFolder] [seed] [expectedVersion] (mapName)");
-			
 		}
 	}
 
 	public void save(String folderName, String mapName, SCMap map, long seed) {
 		try {
 			Path folderPath = Paths.get(folderName);
-			Files.deleteIfExists(folderPath.resolve(mapName));
+			if (Files.exists(folderPath.resolve(mapName))) {
+				Files.walk(folderPath.resolve(mapName))    // Empties the folder in case it exists
```

Comment:
no need to comment, Files.walk.forEach -> Fiels.delete should be obvious to everybody,